spec/models/spree/order_spec.rb

```diff
@@ -195,6 +195,21 @@
       expect(order.updater).to receive(:before_save_hook)
       order.finalize!
     end
+
+    context "extra payments exist that require authorization" do
+      let!(:cash_payment) { build(:payment, state: "completed", amount: order.new_outstanding_balance) }
+      let!(:stripe_payment) { build(:payment, state: "requires_authorization") }
+      before do
+        order.payments << cash_payment
+        order.payments << stripe_payment
+        allow_any_instance_of(Spree::Payment).to receive(:void_transaction!) {}
```

Comment:
Using `any_instance_of` is not ideal; however, putting the allow/expect on the `stripe_payment` itself wasn't working, even though I verified that it is in fact receiving `void_transaction!` - I'm not sure why it wasn't passing.

Reply:
First you allow it to be called and then you expect it to be called? I don't understand. Would it be better to expect it to be called twice or however often it's called?

The problem with any instance is that we don't know if it's voiding the cash payment or the stripe payment. You could create two test cases, one with cash and one with stripe. The first one is not voided the second one should be voided.

Reply:
> First you allow it to be called and then you expect it to be called?

My intention with the `allow` was to pass the empty block so that calling `void_transaction!` doesn't actually try to make any call to Stripe.